### DISCUSSION TITLE
Default random function, how does it work?

### DISCUSSION THREAD

**Discussion by maerobica** [AzuraCast/AzuraCast#3923] [General]
Hello,

we have 3 hour per day playing playlists made of around 2000 songs.
I often see some songs from the playlists are being repeated every day during the week when the same playlists is plaid and some songs that are in the playlist for long time never been plaid. 

Our playlists is General rotation, Default weight (3), song based playlist, random shuffle. Is this random function defined somewhere? how does it work? Is it not effected by listener appearance for instance? if more listeners join the stream during particular song is being plaid, does it not have more chance to be plaid next time same playlists will be streamed compared to other songs? 

thanks

**Comment by BusterNeece** [comment]
@maerobica There are two types of "random" in the system:
 - Random, which is truly random and picks a new song from the playlist every single time with no memory of the previous track; the only thing that will prevent this from being truly random is if you have the "Prevent duplicates" setting enabled.
 - Shuffled, which randomizes the entire playlist once, then plays all the way through it, which should avoid playing the same track multiple times in quick succession. Once the shuffled queue runs out, it reshuffles it and repeats the whole process again.

**Reply by maerobica** [reply]
thank you.

**Comment by maerobica** [comment]
Hello, as our playlist got more songs, our doubts on RANDOM has only increased.

We operate with 35000 songs in 8 playlists. Playlists play around 24 hours divided in periods. so morning playlist (1) ends, afternoon playlist (2) starts and it goes round around the clock... The songs keep on repeating.

We want to ask if folders have any influence on Random function? In other words, if we keep the songs of playlist 1 in different folders: playlist_1_A - 30 tracks , playlist_1_B - 200 tracks, playlist_1_C - 5 tracks... does RANDOM picks the folder first and then the song from that folder? < that might be why we have repeating songs too often. Or does it create a virtual playlist from all the tracks we've uploaded for playlist 1 and then randoms from it? < that is how we imagines it is working.

**Reply by Liam024** [reply]
I would recommend the shuffle function for the many songs.
Random is as described above, just random and it doesn't care if the same song was already playing an hour ago.
But only my recommendation.

**Comment by maerobica** [comment]
To make it more clear:

Playlists type: General rotation.
It is a Song-Based Playlist where Song Playback Order is RANDOM

All 8 playlists have same weight of 3.

It is Standard playlist, shuffles with other standard playlists based on weight.

Include in Automated Assignment is not selected on all playlists.

Avoid Duplicate Artists/Titles is not selected on all playlists because there are different tracks of same artist in all playlists.

**Comment by SpeedyCharly** [comment]
As Slvr Eagle suggested, you should set your playlists to **SHUFFLED** rather than **RANDOM** and enable the **Prevent duplicates** function which prevents same artist/title repeating for a preset period of time within the set playlist. Not sure how it acts on same artist from different playlists...

**Comment by maerobica** [comment]
OK, thanks for the advice, we can try and see the result.

But the question stays. In case we have several folders for same playlist, does the shuffle or the random function first selects one of those folders and then selects particular song to be plaid again in random/shuffle? Or the system just sees all songs from same playlist as track list and chooses randomly or shuffles the song to play, does not matter that all songs are in different folders. Does the folder structure affects random or shuffle functions?

m.

**Reply by SpeedyCharly** [reply]
If you have several folders linked to 1 playlist. the contents of such folders will be seen a a single unit and the whole of it will be either shuffled or randomized as a unit.  Haven't tested that personally, but that is my understanding of how it goes.

**Reply by maerobica** [reply]
yes, us too. we think and understand the same, but what is the true ? :) again, looking for it everywhere
